include/inviwo/core/processors/poolprocessor.h

```diff
@@ -0,0 +1,495 @@
+
+    /**
+     * Cancel all current jobs
+     */
+    void stopJobs();
+
+    /**
+     * Is there any ongoing background jobs
+     */
+    bool hasJobs();
+
+    /**
+     * Dispatch a single background job. The job will be executed in a background thread in
+     * the thread pool. It is important that the job captures its state by value, since it might
+     * outlive the processor. The job can take two optional parameters
+     *     * pool::Stop a stop token to periodically check if the job has been canceled.
+     *     * pool::Progress a callback to report progress of the calculation. The progress is
+     *       represented as a float in the intervall [0.0, 1.0].
+     * If the job takes a progress callback the processor will show a progress bar.
+     * The second functor done is called with the result when the background job is finished. It
+     * will be executed on the main thread, and only if the processor is still valid and the job has
+     * not been stopped. Hence it is safe to refer to the processor in this functor.
+     *
+     * \code{.cpp}
+     * const auto calc = [delay = delay_.get(), image = inport_.getData()]
+     *     ( pool::Stop stop, pool::Progress progress) -> std::shared_ptr<const Image> {
+     *     if (stop) return nullptr;
+     *     auto newImage = std::shared_ptr<Image>(image->clone());
+     *     progress(0.5f);
+     *     // Do some work with the image
+     *     return newImage;
+     * };
+     *
+     * dispatchOne(calc, [this](std::shared_ptr<const Image> result) {
+     *     outport_.setData(result);
+     *     newResults();  // Let the network know that the processor has new results on the
+     *                    // outport.
+     * });
+     * \endcode
+     */
+    template <typename Job, typename Done>
+    void dispatchOne(Job&& job, Done&& done);
+
+    /**
+     * Dispatch a vector of background jobs. The jobs will be executed in a background thread in
+     * the thread pool. It is important that the jobs captures its state by value, since it might
+     * outlive the processor. The jobs can take two optional parameters
+     *     * pool::Stop a stop token to periodically check if the job has been canceled.
+     *     * pool::Progress a callback to report progress of the calculation. The progress is
+     *       represented as a float in the intervall [0.0, 1.0].
+     * If the jobs takes a progress callback the processor will show a progress bar. The progress
+     * from all the jobs will automatically be merged and normalized
+     *
+     * The second functor 'done' is called with the results when the background jobs are finished.
+     * It will be executed on the main thread, and only if the processor is still valid and the jobs
+     * have not been stopped. Hence it is safe to refer to the processor in this functor.
+     *
+     * \code{.cpp}
+     * std::vector<std::function<std::shared_ptr<Mesh>(pool::Stop, pool::Progress progress)>> jobs;
```

Comment:
Could we improve readability hear by using some aliasing, maybe something like:
```c++
template<typename T> using PoolProcessorJob = std::function<std::shared_ptr<T>(pool::Stop, pool::Progress);
```
then we could write `std::vector<PoolProcessorJob<Mesh>> jobs;` instead of `std::vector<std::function<std::shared_ptr<Mesh>(pool::Stop, pool::Progress progress)>> jobs;`

Though, the parameters are optional which would then result in the need of 4 different usings.

Reply:
Since this is all vocabulary types, I think it is good that the reader sees the types to make it easier to understand. Making it short to type here will only make it harder to understand. then you first have to figure out what the aliases mean... also there is no requirement that the return type is a shared_ptr